docs/proposals/config_data.md

```diff
@@ -0,0 +1,316 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named `ConfigData` stored inside `etcd` that
```

Comment:
I am wondering if it should be tightly linked to etcd or be more general as any other kv store e.g. consul. As there is an issue #1957 to add consul. But even beyond that you might want/prefer to store config and other application data in a separate kv store then the use by k8s cluster itself to avoid any bottleneck or unforeseen issue.